src/routers/catalog.ts

```diff
@@ -1,18 +1,27 @@
+import * as Effect from "@effect/io/Effect";
+import { pipe } from "@effect/data/Function";
 import { zodiosRouter } from "@zodios/express";
 import { api } from "../model/generated/api.js";
 import { ApiError, mapCatalogServiceErrorToApiError } from "../model/types.js";
 import { catalogService } from "../services/CatalogService.js";
+import { db, DB } from "../repositories/db.js";
 
 const eservicesRouter = zodiosRouter(api.api);
 
-eservicesRouter.post("/eservices", async (req, res) => {
-  try {
-    await catalogService.createEService(req.body);
-    return res.status(201).end();
-  } catch (error) {
-    const errorRes: ApiError = mapCatalogServiceErrorToApiError(error);
-    return res.status(errorRes.status).json(errorRes).end();
-  }
-});
+eservicesRouter.post("/eservices", (req, res) =>
```

Comment:
Altro esempio di interfacciamento con librerie Promise based (express/axios in questo caso).
Qui ho lavorato 100% in Effect e wrappato il tutto con un `runPromise` alla fine.